packages/block-library/src/video/edit.js

```diff
@@ -97,9 +97,15 @@ function VideoEdit( {
 			setAttributes( { src: undefined, id: undefined } );
 			return;
 		}
+
 		// sets the block's attribute and updates the edit component from the
 		// selected media
-		setAttributes( { src: media.url, id: media.id } );
+		setAttributes( {
+			src: media.url,
+			id: media.id,
+			poster:
+				media.image?.src !== media.icon ? media.image?.src : undefined,
+		} );
```

Comment:
It looks like it `poster` defaults to `undefined`, which is why I picked that -- this avoids having an empty `poster` attribute.

However, when `poster` is "Removed", it's set to empty string: https://github.com/WordPress/gutenberg/blob/009717a3d88fa26cb8613cd64fd987cc8607a210/packages/block-library/src/video/edit.js#L160

This results in an empty `poster` attribute. I'm not sure anything should change about it in this PR, but figured I'd document the finding.

Reply:
Okay, I set this to `undefined` instead in the most recent pass to avoid an empty `poster` attribute on the front-end.